apps/web/components/landing/newsletter.tsx

```diff
@@ -36,28 +35,34 @@ const SubmitButton = () => {
 export function NewsletterForm() {
   const formRef = useRef<HTMLFormElement>(null);
   const { toast } = useToast();
-  return (
-    <form
-      ref={formRef}
-      className="max-w-md"
-      action={async (formData) => {
-        const test = await addEmail(formData);
+  function handleSubmit(event: any) {
```

Comment:
Let's make this typesafe similar to here: https://github.com/unkeyed/unkey/blob/f6174806463e2e96d8120f3cf497de5588ec9987/apps/web/app/(authenticated)/(app)/app/settings/team/invite.tsx#L56